internal/commands/secrets/update_inputs_test.go

```diff
@@ -0,0 +1,83 @@
+package secrets
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+)
+
+func TestResolveSecret(t *testing.T) {
+	testLen := 7
+	secrets := make([]realm.Secret, testLen)
+
+	for i := 0; i < testLen; i++ {
+		secrets[i] = realm.Secret{
+			fmt.Sprintf("secretID%d", i),
+			fmt.Sprintf("secretName%d", i),
+		}
+	}
+
+	for _, tc := range []struct {
+		description    string
+		testInput      string
+		expectedOutput realm.Secret
+	}{
+		{
+			description:    "should return the secret if searching by name",
+			testInput:      "secretName4",
+			expectedOutput: secrets[4],
+		},
+		{
+			description:    "should return the secret if searching by ID",
+			testInput:      "secretID3",
+			expectedOutput: secrets[3],
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			inputs := updateInputs{
+				secret: tc.testInput,
+			}
+			result, err := inputs.resolveSecret(nil, secrets)
+			assert.Nil(t, err)
+			assert.Equal(t, tc.expectedOutput, result)
+		})
+	}
+
+	t.Run("should return an error if we cannot find the secret to update", func(t *testing.T) {
+		inputs := updateInputs{
+			secret: "cannotFind",
+		}
+
+		result, err := inputs.resolveSecret(nil, secrets)
+		assert.Equal(t, realm.Secret{}, result)
```

Comment:
when asserting `error` return values, we don't need to assert anything on the first return value

and just to avoid any potential `nil` panics with `err.Error()`, try to use the `assert.Equal` with an actual error var (`errors.New("unable to find secret: cannotFind"` in this case)